src/front/spv/mod.rs

```diff
@@ -2761,7 +2761,10 @@ impl<I: Iterator<Item = u32>> Parser<I> {
                     break None;
                 }
                 Op::BranchConditional => {
-                    inst.expect_at_least(4)?;
+                    let weighted = inst.wc == 6;
```

Comment:
perhaps we could refactor it slightly differently?
I.e. it already used to check `expect_at_least(4)`. What it was missing though is eating those additional words.
So we could have just added something like this to the end:
```rust
for _ in 4 .. inst.wc {
  let _ = self.next()?;
}
```

Reply:
The way I did it in this PR adds a little bit of validation, the spirv spec only allows 6 words (with weights) or 4 (without weights).

Your solution would allow for 5 words and 7+ words inputs. If you think this kinda of validation hurts more than it helps then I'm okay with your proposal.

Reply:
We aren't in the business of validating input SPIR-V. That would blow up the complexity of spv-in rather significantly.
We talked previously about whether or not GLSL-in needs to be validated, and I argued against it. But a strong point for GLSL is that it's human-writable, therefore even limited validation is useful.
SPIR-V though is not human writable, so we expect it to be generated correctly in the first place.